api/v1alpha4/awsmachine_types.go

```diff
@@ -66,43 +95,64 @@ type AWSMachineSpec struct {
 	// +optional
 	AdditionalSecurityGroups []AWSResourceReference `json:"additionalSecurityGroups,omitempty"`
 
-	// AvailabilityZone is references the AWS availability zone to use for this instance.
-	// If multiple subnets are matched for the availability zone, the first one return is picked.
-	// +optional
-	AvailabilityZone *string `json:"availabilityZone,omitempty"`
+	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Cluster API.
+	// For this infrastructure provider, the ID is equivalent to an AWS Availability Zone.
+	// If multiple subnets are matched for the availability zone, the first one returned is picked.
+	FailureDomain *string `json:"failureDomain,omitempty"`
 
 	// Subnet is a reference to the subnet to use for this instance. If not specified,
 	// the cluster subnet will be used.
 	// +optional
 	Subnet *AWSResourceReference `json:"subnet,omitempty"`
 
-	// SSHKeyName is the name of the ssh key to attach to the instance.
-	SSHKeyName string `json:"sshKeyName,omitempty"`
+	// SSHKeyName is the name of the ssh key to attach to the instance. Valid values are empty string (do not use SSH keys), a valid SSH key name, or omitted (use the default SSH key name)
+	// +optional
+	SSHKeyName *string `json:"sshKeyName,omitempty"`
+
+	// RootVolume encapsulates the configuration options for the root volume
+	// +optional
+	RootVolume *Volume `json:"rootVolume,omitempty"`
 
-	// RootDeviceSize is the size of the root volume in gigabytes(GB).
+	// Configuration options for the non root storage volumes.
 	// +optional
-	RootDeviceSize int64 `json:"rootDeviceSize,omitempty"`
+	NonRootVolumes []*Volume `json:"nonRootVolumes,omitempty"`
```

Comment:
Same comment as above.
Did not do any changes on the v1alpha4 types.
Let's track the changes we want to have in v1alpha4 as separate issues so that it will be easier to review/revert.